### PR TITLE
fix: get api host from middleware url

### PR DESCRIPTION
**Pull request by dawid-ziobro**
## Description

Base url for middleware is no longer computed and is not based on headers anymore. 
Now `middlewareUrl` is used instead and this config option is now required.
Configuration in `nuxt.config.js` has to be extended.

## Related Issue

CT-208

## Motivation and Context

Headers can be overridden which can lead to security issues.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

#### Changelog
- [ ] I have updated the Changelog ([V1](https://github.com/DivanteLtd/vue-storefront/blob/develop/CHANGELOG.md)) [v2](https://docs-next.vuestorefront.io/contributing/creating-changelog.html) and mentioned all breaking changes in the public API.
- [ ] I have documented all new public APIs and made changes to existing docs mentioning the parts I've changed so they're up to date.

#### Tests
- [x] I have written test cases for my code
- [ ] I have tested my Pull Request on production build and (to my knowledge) it works without any issues
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

#### Code standards
- [ ] My code follows the code style of this project.

- [ ] I have followed [naming conventions](https://github.com/kettanaito/naming-cheatsheet)

#### Docs
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: packages/core/docs/changelog/6680.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+module.exports = {
+  description: 'get api host from middleware url',
+  link: 'https://github.com/vuestorefront/vue-storefront/pull/6680',
+  isBreaking: true,
+  breakingChanges: [
+    'middlewareUrl has to be configured for nuxt. Please follow instruction in migration guide for version 2.5.7.'
```
Comment (filipsobol) [inline packages/core/docs/changelog/6680.js:6]:
```suggestion
    'The `middlewareUrl` property in the `nuxt.config.js` file is now required. Please follow the instruction in the Migration Guide.'
```

File: packages/core/docs/getting-started/configuration.md [hunk 2]
```diff
@@ -16,6 +16,21 @@ You can learn more about this file and available configuration options on the [N
 
 The `middleware.config.js` file is as essential as `nuxt.confis.js`, but much simpler and likely smaller. It configures the Server Middleware used for communication with e-commerce platforms and contains sensitive credentials, custom endpoints, queries, etc.
 
+What is common for every integration is setting `middlewareUrl`. This is required and have to point to middleware. Usually it is your application domain.
```
Comment (filipsobol) [inline packages/core/docs/getting-started/configuration.md:19]:
```suggestion
What is typical for every integration is the `middlewareUrl` property which defines the URL to the Server Middleware. Usually, it's your application domain followed by the `/api` path.
```

File: packages/core/docs/getting-started/configuration.md [hunk 3]
```diff
@@ -16,6 +16,21 @@ You can learn more about this file and available configuration options on the [N
...
+```javascript
+// nuxt.config.js
+export default {
+  publicRuntimeConfig: {
+    middlewareUrl: 'https://yourdomain.com/api/'
+  }
+}
+```
+
+For development just use `http://localhost:3000/api`.
```
Comment (filipsobol) [inline packages/core/docs/getting-started/configuration.md:32]:
```suggestion
For the local development, set it to `http://localhost:3000/api`.
```

File: packages/core/docs/reference/migrate/2.5.7/overview.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
+# Migrating projects to 2.5.7
+
+## Extend `nuxt.config.js`
+
+In this release we've removed generating `baseUrl` for api calls, based on request headers.
+Now it has to be configured and proper `middlewareUrl` needs to be passed.
+
+To do it, just extend `nuxt.config.js` and add `middlewareUrl`.
+This field is now required, before it was optional.
```
Comment (filipsobol) [inline packages/core/docs/reference/migrate/2.5.7/overview.md:9]:
```suggestion
## Update `nuxt.config.js`

In this release, we made the `middlewareUrl` property required for security reasons. Open the  `nuxt.config.js` file and add the `middlewareUrl` property like shown below:
```

File: packages/core/docs/reference/migrate/2.5.7/overview.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+```javascript
+// nuxt.config.js
+export default {
+  publicRuntimeConfig: {
+    middlewareUrl: 'https://yourdomain.com/api/'
```
Comment (filipsobol) [inline packages/core/docs/reference/migrate/2.5.7/overview.md:15]:
```suggestion
    middlewareUrl: 'https://yourdomain.com/api/' // For the local development, set it to `http://localhost:3000/api/`.
```

File: packages/core/docs/reference/migrate/2.5.7/overview.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+}
+```
+
+:::warning
+Make sure to pass whole url with protocol and/or port and suffix it with `/api/`.
+:::
+
+For development, you can just use `http://localhost:3000/api/`.
+
+If you don't want to hardcode url in config file, you can use environmental variables.
```
Comment (filipsobol) [inline packages/core/docs/reference/migrate/2.5.7/overview.md:26]:
```suggestion

If you don't want to hardcode the URL in the configuration file, you can use environmental variables.
```

File: packages/core/docs/reference/migrate/2.5.7/overview.md [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+```javascript
+// nuxt.config.js
+export default {
+  publicRuntimeConfig: {
+    middlewareUrl: process.env.API_BASE_URL
+  }
+}
+```
+
+And then add entry in `.env` file or use any other method for passing environmental variables that suits your needs.
```
Comment (filipsobol) [inline packages/core/docs/reference/migrate/2.5.7/overview.md:39]:
```suggestion
Then add an entry in the `.env` file or use any other method for passing environmental variables that suits your needs.
```

**Comment by lsliwaradioluz** [comment]
Should we also update this part of the VSF docs about [separating server-middleware](https://docs.vuestorefront.io/v2/architecture/server-middleware.html#separating-server-middleware-from-nuxt-js)?

image

**Comment by dawid-ziobro** [comment]
@lsliwaradioluz i don't think so, nothing changes for this case.

**Review by lukasborawski — APPROVED** [review]

**Pull request merged**